src/pygeom.c

```diff
@@ -79,7 +85,14 @@ static PyObject *GeometryObject_ToWKT(GeometryObject *obj, char *format)
 
 static PyObject *GeometryObject_repr(GeometryObject *self)
 {
-    return GeometryObject_ToWKT(self, "<pygeos.Geometry %s>");
+    PyObject *result = GeometryObject_ToWKT(self, repr_fmt);
+    // we never want a repr() to fail; that can be very confusing
+    if (result == NULL) {
+        PyErr_Clear();
+        return PyUnicode_FromFormat(repr_fmt, "Exception in WKT writer");
+    } else {
```

Comment:
minor suggestion; a bare return would be fine here too, no need for conditions since both branches return:

```
if (result == NULL) { ... }

return result
```